### PR TITLE
[4.0] Add TFA status to users list

### PR DESCRIPTION
**Pull request by brianteeman**
I wrote this for a client on J3 but as J3 is not accepting new features I am submitting it for J4. There is currently no way to see if a user is actually using two factor authentication. This resolves that.

image

- If there are no TFA plugins enabled then you wont see any changes
- If there are TFA plugins enabled then you will see a new column in the Users list
- If the user is using TFA then they will have the standard green tick
- If the user is not using TFA then they will have the standard grey cross

### CODE REVIEW

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 5474693028c71e2417d946b260b111db48f9294d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23900">issues.joomla.org/tracker/joomla-cms/23900</a>.</sub>

File: administrator/components/com_users/tmpl/users/default.php [hunk 1]
```diff
@@ -66,6 +69,11 @@
 								<th scope="col" style="width:5%" class="text-center d-none d-md-table-cell">
 									<?php echo HTMLHelper::_('searchtools.sort', 'COM_USERS_HEADING_ACTIVATED', 'a.activation', $listDirn, $listOrder); ?>
 								</th>
+								<?php if ($tfa) : ?> 
+								<th scope="col" style="width:5%" class="text-center d-none d-md-table-cell">
+									<?php echo Text::_('COM_USERS_HEADING_TFA'); ?>
+								</th>
```
Comment (Quy) [inline administrator/components/com_users/tmpl/users/default.php]:
```suggestion
								</th>
```

File: administrator/components/com_users/tmpl/users/default.php [hunk 2]
```diff
@@ -150,6 +158,17 @@
...
 								</td>
+								<?php if ($tfa) : ?> 
+								<td class="text-center d-none d-md-table-cell tbody-icon">
+									<?php if (!empty($item->otpKey)) : ?>
+										<span class="icon-publish" aria-hidden="true"></span>
+										<span class="sr-only"><?php echo Text::_('COM_USERS_TFA_ACTIVE'); ?></span>
+									<?php else : ?> 
+										<span class="icon-unpublish" aria-hidden="true"></span>
+										<span class="sr-only"><?php echo Text::_('COM_USERS_TFA_NOTACTIVE'); ?></span>
+									<?php endif ; ?>
```
Comment (Quy) [inline administrator/components/com_users/tmpl/users/default.php]:
```suggestion
									<?php endif ; ?>
```

Comment (Quy) [inline administrator/components/com_users/tmpl/users/default.php]:
```suggestion
									<?php endif; ?>
```

**Comment by brianteeman** [comment]
No need for new tests. The changes were just white space

**Comment by zero-24** [comment]
The old discussion for referenz: https://github.com/joomla/joomla-cms/pull/12082

**Comment by brianteeman** [comment]
I forgot about that ;) Would have saved me some time.

Reading through all the comments there I believe they are all addressed here

**Comment by HLeithner** [comment]
I may missed the reason in the other PR, could you please give me a summary?

**Comment by zero-24** [comment]
> I may missed the reason in the other PR, could you please give me a summary?

What reason? That it got closed? I had closed it because it got ignored and by that time i moved it to j-over so it would not be lost. Feel free to include it into 3.9.4 :-)

**Comment by brianteeman** [comment]
It's a new feature so I submitted it J4 as J3 is not accepting new features.

**Comment by zero-24** [comment]
Sorry yes got confused.

**Comment by euismod2336** [comment]
I have tested this item :white_check_mark: successfully on 9e62e78bc7662f8d51836c48818baca33a50a0ec<br /><br />works as advertised<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23900">issues.joomla.org/tracker/joomla-cms/23900</a>.</sub>

**Comment by ghost** [comment]
@ChristineWk can you please retest?

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 9e62e78bc7662f8d51836c48818baca33a50a0ec<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23900">issues.joomla.org/tracker/joomla-cms/23900</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by SniperSister** [comment]
Merged, thanks @brianteeman!

**Comment by brianteeman** [comment]
thanks

**Pull request merged**